linkedList.c

```diff
@@ -0,0 +1,237 @@
+    }
+
+    (*node)->value = value;
+    (*node)->next = NULL;
+
+    return SUCCESS;
+}
+
+/*
+Create first element of a list.
+
+@param list - list to add element to.
+@param value - value of the to-be first element.
+
+@return LinkedListError with exit code.
+*/
+LinkedListError addFirstNode(LinkedList* list, int value) {
+    return initNode(&(list->head), value);
+}
+
+LinkedListError append(LinkedList* list, int value) {
+    Node* iterator = list->head;
+
+    if (iterator == NULL) {
+        return addFirstNode(list, value);
+    }
+
+    while (iterator->next != NULL) {
+        iterator = iterator->next;
+    }
+
+    return initNode(&(iterator->next), value);
+}
+
+LinkedListError prepend(LinkedList* list, int value) {
+    Node* tmp = list->head;
+
+    if (tmp == NULL) {
+        return addFirstNode(list, value);
+    }
+
+    LinkedListError exit_value = initNode(&(list->head), value);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    list->head->next = tmp;
+
+    return SUCCESS;
+}
+
+/*
+Points the given pointer to the element at the given index of the given list.
+
+@param list - given list.
+@param index - given index.
+@param pointer - given pointer.
+
+@return LinkedListError with exit code.
+*/
+LinkedListError getNodeAt(LinkedList* list, size_t index, Node** pointer) {
+    Node* iterator = list->head;
+
+    if (iterator == NULL) {
+        return INDEX_ERROR;
+    }
+
+    for (int i = 0; i < index; i++) {
+        if (iterator->next == NULL) {
+            return INDEX_ERROR;
+        }
+
+        iterator = iterator->next;
+    }
+
+    *pointer = iterator;
+    return SUCCESS;
+}
+
+LinkedListError insert(LinkedList* list, size_t index, int value) {
+    Node* iterator;
+
+    if (index == 0) {
+        return addFirstNode(list, value);
+    }
+    
+    LinkedListError exit_value = getNodeAt(list, index - 1, &iterator);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    Node* tmp;
+    exit_value = initNode(&tmp, value);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    tmp->next = iterator->next;
+    iterator->next = tmp;
+
+    return SUCCESS;
+}
+
+/*
+Removes the first node of a list.
+
+@param list - list to remove node from.
+
+@return LinkedListError with exit code.
+*/
+LinkedListError removeFirstNode(LinkedList* list) {
+    Node* tmp = list->head->next;
+
+    if (tmp == NULL) {
+        return INDEX_ERROR;
+    }
+
+    free(list->head);
+    list->head = tmp;
+
+    return SUCCESS;
+}
+
+LinkedListError removeAt(LinkedList* list, size_t index) {
+    Node* iterator;
+
+    if (index == 0) {
+        return removeFirstNode(list);
+    }
+
+    LinkedListError exit_value = getNodeAt(list, index - 1, &iterator);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    if (iterator->next == NULL) {
+        return INDEX_ERROR;
+    }
+
+    Node* tmp = iterator->next->next;
+    free(iterator->next);
```

Comment:
Perhaps create a local variable for `iterator->next` in this code part to make it more readable, such as `node_to_remove`